### PR TITLE
Address comments from discussion in the WG

### PR DESCRIPTION
**Pull request by tireddy2**

### CODE REVIEW

**Review by OR13 — APPROVED** [review]

**Comment by OR13** [comment]
Lets merge this as is, I will update the implementation to match it, and we can start the next cycle of review.

File: draft-rha-jose-hpke-encrypt.md [hunk 1]
```diff
@@ -142,29 +142,32 @@ This specification uses the following abbreviations and terms:
...
+  
+  *  HPKE in a multiple recipient setup referred to as Key Encryption mode. In this case, the shared secret established through the HPKE will wrap the CEK.
+
+In both cases a new JOSE header parameter, called 'ek', is used to convey the content of the "enc" structure defined in the HPKE specification. "enc" represents the serialized public key.
 
-The 'encapsulated_key' parameter contains the encapsulated key, which is output of the HPKE KEM, and is represented as a base64url encoded string. The parameter "kty" MUST be present and set to "OKP" defined in Section 2 of {{RFC8037}}.
+When the alg value is set to any of algorithms registered by this specification then the 'ek' header parameter MUST be present.
 
-### HPKE Usage in Direct and Key Agreement with Key Wrapping
+The 'ek' parameter contains the encapsulated key, which is output of the HPKE KEM, and is represented as a base64url encoded string. The parameter "kty" MUST be present and set to "OKP" defined in Section 2 of {{RFC8037}}.
```
[the remark this answers is not in the document]
Comment (tireddy2) [reply draft-rha-jose-hpke-encrypt.md:158]: Good point,  I will remove the line.

File: draft-rha-jose-hpke-encrypt.md [hunk 2]
```diff
@@ -142,29 +142,32 @@ This specification uses the following abbreviations and terms:
...
 
-#### HPKE Usage in Key Agreement with Key Wrapping mode
+#### HPKE Usage in Key Encryption mode
 
-In the JWE JSON Serialization, the sender MUST place the 'encapsulated_key' parameter in the per-recipient unprotected header. Optionally, the per-recipient unprotected header MAY contain the 'kid' parameter used to identify the static recipient public key used by the sender.
+In Key Encryption mode, the sender MUST place the 'ek' and 'alg' parameters in the per-recipient unprotected header to indicate the use of HPKE. Optionally, the per-recipient unprotected header MAY contain the 'kid' parameter used to identify the static recipient public key used by the sender. In this mode, JWE serialization MUST be used. In this setup, the 'enc' (Encryption Algorithm) parameter MUST be present to identify the content encryption algorithm used to perform encryption on the plaintext to produce the ciphertext.
```
[the remark this answers is not in the document]
Comment (tireddy2) [reply draft-rha-jose-hpke-encrypt.md:170]: I don't get the comment. As per the discussion in the WG, JWE serialization must be used for Key Encryption mode.

[the remark this answers is not in the document]
Comment (tireddy2) [reply draft-rha-jose-hpke-encrypt.md:170]: Good point, I will add the following text:
The "enc" Header Parameter MUST be integrity protected and it MUST occur only within JWE Protected Header.

**Pull request merged**